src/librustc/hir/lowering.rs

```diff
@@ -2540,7 +2583,7 @@ impl<'a> LoweringContext<'a> {
 
                 let kind = hir::GenericParamKind::Type {
                     default: default.as_ref().map(|x| {
-                        self.lower_ty(x, ImplTraitContext::disallowed())
+                        self.lower_ty(x, ImplTraitContext::Existential(None))
```

Comment:
I believe that's handled by the separate `lower_generics`... similarly named, so it can easily throw one off, but the two code pathways shouldn't intersect in this scenario.